### PR TITLE
Ensure that EB and datacarrier size are set properly after a node restart

### PR DESCRIPTION
**Pull request by ptschip**
After the fork if a node restarts we need to make sure that the settings
are properly enabled during intitialization.

### CODE REVIEW

**Comment by ptschip** [comment]
I just tested this on standalone regtest.  I does work...after forktime is active and  median time is passed and we have activated the fork then when we restart we get the correct MG/EB and OP_return sizes and also the QT settings for MB/EB are updated appropriately.

**Comment by ptschip** [comment]
@gandrewstone ok this is ready for review

File: src/utiltime.cpp [hunk 1]
```diff
@@ -22,7 +22,7 @@ int64_t GetTime()
     if (nMockTime)
         return nMockTime;
 
-    time_t now = time(NULL);
+    time_t now = time(nullptr);
```
Comment (sickpig) [inline src/utiltime.cpp:25]: nit: this change seems unrelated.

File: qa/rpc-tests/may152018-forkactivation_1.py [hunk 2]
```diff
@@ -34,7 +34,8 @@ def setup_network(self, split=False):
     def run_test(self):
 
         # Advance the time beyond the timeout value
-        cur_time = int(time.time())
+        #cur_time = int(time.time())
+        cur_time = 1526400000 - 10
```
Comment (sickpig) [inline qa/rpc-tests/may152018-forkactivation_1.py:38]: what's wrong with using `time.time()`?

**Review by sickpig — APPROVED** [review]

File: src/init.cpp [hunk 3]
```diff
@@ -1131,6 +1131,19 @@ bool AppInit2(Config &config, boost::thread_group &threadGroup, CScheduler &sche
         mempool.ReadFeeEstimates(est_filein);
     fFeeEstimatesInitialized = true;
 
+    // Set the EB/MG and datacarrier size if we have restarted after the fork has already happened.
+    // It is possible that we need to override the old settings in QT and bitcoin.conf.
+    if (IsMay152018Enabled(chainparams.GetConsensus(), chainActive.Tip()))
+    {
+        // Bump the accepted block size to 32MB and the default generated size to 8MB
```
Comment (sickpig) [inline src/init.cpp]: comment above said we are going to set max default generated size (`maxGeneratedBlock`)  to 8MB which is not the case.

**Comment by sickpig** [comment]
PR is ok but I think we should fix the minor nits raised in the comments before merging.

File: src/utiltime.cpp (hunk 1 above)
Comment (ptschip) [reply src/utiltime.cpp:25]: i having trouble with mocktime and looked through the code and i thought why not, just one change in that cpp.

File: qa/rpc-tests/may152018-forkactivation_1.py (hunk 2 above)
Comment (ptschip) [reply qa/rpc-tests/may152018-forkactivation_1.py:38]: I was having trouble with -mocktime in the new set of tests.  I suppose I could have launched here with the tweak of forktime but then i would have had to rework upper set of tests ... (as simple as they look it actual took me quite a while to get the new tests to work and didn't want to go back over and edit them - i'll look at this again later when i get back).

File: src/init.cpp (hunk 3 above)
Comment (ptschip) [reply src/init.cpp]: done

File: src/utiltime.cpp (hunk 1 above)
Comment (sickpig) [reply src/utiltime.cpp:25]: fair enough

File: qa/rpc-tests/may152018-forkactivation_1.py (hunk 2 above)
Comment (sickpig) [reply qa/rpc-tests/may152018-forkactivation_1.py:38]: fair enough.

maybe we could merge as is while adding  a `TODO` on the code so that we won't lose track of it.
